src/gzchunked/mod.rs

```diff
@@ -0,0 +1,210 @@
+
+        assert_eq!([1, 2, 3, 4], decoder.try_next_data().unwrap().as_slice());
+        assert_eq!([] as [u8; 0], decoder.try_next_data().unwrap().as_slice());
+        assert_eq!([1, 2, 3, 4, 5, 6, 7, 8], decoder.try_next_data().unwrap().as_slice());
+        assert_eq!([1], decoder.try_next_data().unwrap().as_slice());
+        assert!(decoder.try_next_data().is_none());
+    }
+
+    #[test]
+    fn test_encode_and_decode_one_per_block() {
+        let mut encoder = GzChunkedEncoder::new(GzChunkedCompression::default());
+        let mut decoder = GzChunkedDecoder::new();
+
+        encoder.write(&[1, 2, 3, 4]).unwrap();
+        let encoded_block = encoder.next_chunk().unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+        assert_eq!([1, 2, 3, 4], decoder.try_next_data().unwrap().as_slice());
+        assert!(decoder.try_next_data().is_none());
+
+        encoder.write(&[]).unwrap();
+        let encoded_block = encoder.next_chunk().unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+        assert_eq!([] as [u8; 0], decoder.try_next_data().unwrap().as_slice());
+        assert!(decoder.try_next_data().is_none());
+
+        encoder.write(&[1, 2, 3, 4, 5, 6, 7, 8]).unwrap();
+        let encoded_block = encoder.next_chunk().unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+        assert_eq!([1, 2, 3, 4, 5, 6, 7, 8], decoder.try_next_data().unwrap().as_slice());
+        assert!(decoder.try_next_data().is_none());
+
+        encoder.write(&[1]).unwrap();
+        let encoded_block = encoder.next_chunk().unwrap();
+        decoder.write(encoded_block.as_slice()).unwrap();
+        assert_eq!([1], decoder.try_next_data().unwrap().as_slice());
+        assert!(decoder.try_next_data().is_none());
+    }
+
+    #[test]
+    fn test_encode_and_decode_random() {
+        let mut rng = StdRng::seed_from_u64(20200509);
+        let mut encoder = GzChunkedEncoder::new(GzChunkedCompression::default());
+        let mut decoder = GzChunkedDecoder::new();
+        let mut expected_data: Vec<Vec<u8>> = Vec::new();
+        let mut decoded_data: Vec<Vec<u8>> = Vec::new();
+        for _ in 0..256 {
+            let size = rng.gen_range(128, 256);
+            let data: Vec<u8> = (0..size).map(|_| rng.gen()).collect();
+            encoder.write(data.as_slice()).unwrap();
+            if let Some(block) = encoder.try_next_chunk().unwrap() {
+                decoder.write(block.as_slice()).unwrap();
+                while let Some(data) = decoder.try_next_data() {
+                    decoded_data.push(data);
+                }
+            }
+            expected_data.push(data);
+        }
+        let last_block = encoder.next_chunk().unwrap();
+        decoder.write(last_block.as_slice()).unwrap();
+        while let Some(data) = decoder.try_next_data() {
+            decoded_data.push(data);
+        }
+        assert_eq!(expected_data, decoded_data);
+    }
+}
```

Comment:
I would also add (very simple) tests that encoding with non-default (lowest and highest, preferably) compression level works as expected—it should, because this just delegates to `flate2` for now, but tests are supposed to guard you against breakages caused by implementation changes.